docs/2_Step_Types.md

```diff
@@ -0,0 +1,77 @@
+# Step Types
+
+There are several built-in types of steps:
+
+### Source Code Management(SCM)
+
+Source Code Management step is meant to be the first step of a pipeline. When you add a new pipeline, you are required to set up the Souce Code Management step.
+
+In SCM step, Choose a authenticated account( for how to do authorization please refer to [Source Code Management Integration](./3_Source_Code_management.md)) and select a repository, fill the **Branch** field to specify which git branch to use.
+
+If you enable the **Webhook** option, it means that the pipeline can be triggered by a push event webhook. For example if you are using Github here, a Github webhook will be automatically created and once a push happens in relevant repository & branch, Github will send a POST request to Rancher server and trigger a new run of the pipeline.
+
+### Build
+
+Build step is for building a docker image. There are normally two cases of building an image in continuous integration using docker. 
+
+1. Build a context image to be used in later tasks such as testing.
+2. Build an image as the artifact of the CI process, which can be used in later testing and deployment.
+
+You can use a Dockerfile in your source code repository or upload and configure one which is excluded from the repository. You can configure the **Build Path** and **Image Tag** of the built image. The image tag should be the full image name containing image registry prefix.
+
+To push the built image to a registry, simply click and enable the **push** option. Rancher Pipeline uses [registry credentials](http://rancher.com/docs/rancher/latest/en/environments/registries/) which are stored in Rancher server. If related registry credential is not configured yet, The UI will notice and guide you there.
+
+### Task
+
+Task step is for running arbitrary shell commands in an arbitrary context image. For example, running PHPUnit tests using the php image or running a `go test` using the golang image. You can specify a context image in **Image** field. To use previously referenced images in the pipeline, you can click dropdown button on the right and select one of them.
+
+When you input some commands in the **Command** textarea, these command will be wrapped in a file and executed by `/bin/sh`. To run with a custom entrypoint and commands, you can click **Custom Entrypoint** and configure them according to your need.
```

Comment:
1. these **commands**
2. I can just configure `custom entrypoint` in **Custom Entrypoint**, right?